FerramAerospaceResearch/FARGUI/FAREditorGUI/Simulation/StabilityDerivCalculator.cs

```diff
@@ -292,8 +316,8 @@ public StabilityDerivOutput CalculateStabilityDerivs(double u0, double q, double
             pertOutput.Cd = (pertOutput.Cd - nominalOutput.Cd) / 0.1;
             pertOutput.Cm = (pertOutput.Cm - nominalOutput.Cm) / 0.1;
 
-            pertOutput.Cl *= q * area / mass;
-            pertOutput.Cd *= q * area / mass;
+            pertOutput.Cl *= -q * area / mass; // Rodhern: Replaced 'q' by '-q', so that formulas
+            pertOutput.Cd *= -q * area / mass; //  for Ze and Xe match those for Zu and Xu.
```

Comment:
I'd be careful about changing signs, I'm sure ferram had a good reason to set them the way they were. If anything, the negative signs shouldn't be there, the equations follow https://courses.cit.cornell.edu/mae5070/DynamicEquations.pdf (for the most part)

Reply:
You can see my reason for changing signs here: [Kerbal Space Program Forum](https://forum.kerbalspaceprogram.com/index.php?/topic/109098-official-far-craft-repository/&do=findComment&comment=2425057). It would be great if you can point out where it is that my reasoning fails.

Reply:
You're probably right, at the very least the signs should be consistent for the same axes. I'm not sure about the need for https://github.com/dkavolis/Ferram-Aerospace-Research/blob/d9665fba79aa90b32226a5e60a95c3540b35109b/FerramAerospaceResearch/FARGUI/FAREditorGUI/Simulation/StabilityDerivCalculator.cs#L235-L236
according to the equations.

Reply:
I have not really paid attention to that. Let us run through some quick back of the envelope calculations and verify that it makes sense to us. Let us look at Zw (i.e. dZ/dw).

The initial state is the aircraft flying horizontally at speed u0. In fact moving at speed u = u0 (forward), w = 0 (downward) and v = 0 (rightward). The angle of attack is some alpha.

Now imagine that we manipulate the direction of travel by rotating it ever so slightly towards the ground. We are in effect increasing alpha, and this will cause two changes that we will monitor. It changes the velocity vector and it changes the downward force. Say the rotation amount is tau radians. As an example of a specific near zero numerical value of tau I will use the symbol 2°.

The velocity deltas are,
 dw = sin(tau) * u0 - 0 ≈ 2° * u0,
 du = cos(tau) * u0 - u0 ≈ 0,
 dv = 0.

The downward force delta (or in our case actually the acceleration delta) is, for some relevant constant k,
 dZ = -cos(tau) * k * u0² * CL(AoA= alpha + tau)  // the new lift component
    + -sin(tau) * k * u0² * CD(AoA= alpha + tau)  // the new drag component
    - -cos(  0) * k * u0² * CL(AoA= alpha +   0)  // the old downward force
  ≈ tau * d( -cos(tau) * k * u0² * CL(tau) + -sin(tau) * k * u0² * CD(tau) )
  = tau * k * u0² * ((-cos * dCL) + (sin * CL) + (-sin * dCD) + (-cos * CD))
  ≈ 2° * k * u0² * (-dCL + -CD)

You probably guessed by now that CL(tau) is the coefficient of lift in the given circumstances at an angle of attack equal to alpha plus tau, that CD(tau) is the corresponding coefficient of drag, and that dCL and dCD are the (mathematical) derivatives of CL and CD with respect to the angle of attack parameter (in radians).

Now if q = 1/2 * density * u0² and k = 1/2 * density * area / mass we get
 Zw = dZ/dw ≈ 2° * k * u0² * (-dCL + -CD) / (2° * u0)
  = -k * u0² * (dCL + CD) / u0
  = -(1/2 * density * area / mass) * u0² * (dCL + CD) / u0
  = -q * area * (dCL + CD) / (mass * u0)

Now compare to the code:
 dCL: pertOutput.Cl = (pertOutput.Cl - nominalOutput.Cl) / (2 * FARMathUtil.deg2rad);
 (dCL + CD): pertOutput.Cl += nominalOutput.Cd;
 -q * area * (dCL + CD) / (mass * u0): pertOutput.Cl *= -q * area / (mass * u0);

I think it makes sense. Please point out mistakes that I make so that we do not end up confusing other comment readers.